tests/scripts/all.sh

```diff
@@ -1024,6 +1024,43 @@ component_check_test_cases () {
     unset opt
 }
 
+component_check_test_dependencies () {
+    msg "Check: test case dependencies: legacy vs PSA" # < 1s
+
+    found="check-test-deps-found-$$"
+    expected="check-test-deps-expected-$$"
+
+    # Find legacy dependencies in PSA tests
+    grep 'depends_on' tests/suites/test_suite_psa* |
```

Comment:
The wildcard pattern might include things we don't want in developer scripts, such as editor backup files.

```suggestion
    grep 'depends_on' tests/suites/test_suite_psa*.data tests/suites/test_suite_psa*.function |
```